### ISSUE TITLE
Improve highlighting for rules relying on eslint core

### ISSUE THREAD

**Issue by quentin-jaquier-sonarsource** [SonarSource/SonarJS#1926]
When we use a rule from eslint core, we don't control which node is reported, in the case of big nodes, the resulting highlighting is ugly in the UI. Knowing that 30 more rules (see https://github.com/SonarSource/SonarJS/issues/1918) will rely on eslint-core, this situation may happen again in the future.

### Rules concerned

- S138 (already fixed in https://github.com/SonarSource/SonarJS/issues/1750)
- S3498 (based on [object-shorthand](https://eslint.org/docs/rules/object-shorthand))

If only a few rules are impacted, implementing our own rule seems the best option.
If we realize that a lot of rules are impacted, we should find a way to re-use eslint implementation, and modify the node reported, (by overwriting the report function for example). 
We could take inspiration from typescript-eslint, doing something similar for all rules listed [here](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/typings/eslint-rules.d.ts).

**Feel free to comment this ticket if you notice other ugly highlighting.**

**Comment by jblievremont** [comment]
For the record, here is how it looks like [in SonarCloud](https://sonarcloud.io/code?id=sonarlint-connected-mode-test&selected=sonarlint-connected-mode-test%3Asrc%2Fmain%2Fjs%2Fmain.js):

![image](https://user-images.githubusercontent.com/1734254/72800109-e2487c80-3c46-11ea-8ee5-94d7d1d85e9a.png)

And in SonarLint for VS Code:

![image](https://user-images.githubusercontent.com/1734254/72800135-eb394e00-3c46-11ea-8693-a26a579d2e82.png)

**Comment by p3k** [comment]
i’d like to chime in with some more cases where the underlining should not continue across a whole function; here is the test case:

```js
var foo = () => {
  const bar = () => {
    void 0;
    void 0;
    void 0;
    void 0;
    void 0;
  };

  return bar;
};
```

**Unexpected var, use let or const instead.**

![image](https://user-images.githubusercontent.com/16631/74839310-6022a500-5325-11ea-9bcb-7c7d7fea4a6a.png)

If I replace `var` with `const` a similar underlining profusion happens for another issue:

**Immediately return this expression instead of assigning it to the temporary variable "bar".**

![image](https://user-images.githubusercontent.com/16631/74839553-bc85c480-5325-11ea-9ab6-379046df9d0a.png)

Could maybe underlining of functions generally be reduced to one line per function?

**Comment by vilchik-elena** [comment]
closing. we will create an issue for each rule requiring location improvement
